pandas/tests/dtypes/cast/test_promote.py

```diff
@@ -163,9 +163,160 @@ def _assert_match(result_fill_value, expected_fill_value):
     assert match_value or match_missing
 
 
-def test_maybe_promote_int_with_int():
-    # placeholder due to too many xfails; see GH 23982 / 25425
-    pass
+@pytest.mark.parametrize(
+    "dtype, fill_value, expected_dtype",
+    [
+        # size 8
+        ("int8", 1, "int8"),
+        ("int8", np.iinfo("int8").max + 1, "int16"),
```

Comment:
I guess for comprehensiveness should we also add unsigned `fill_value` values for a signed `dtype`? Looks like we have the other way around

Reply:
Makes sense, but for the moment this is porting tests from one of @h-vetinari’s PRs with minimal changes.